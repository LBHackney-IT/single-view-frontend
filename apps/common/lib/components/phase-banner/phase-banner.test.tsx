import React from "react";

import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";

import { PhaseBanner } from ".";

test("it renders a default blue variant of the Phase Banner", async () => {
  const { container } = render(
    <PhaseBanner tag="DEV">
      <span>Development Environment</span>
    </PhaseBanner>,
  );

  expect(container).toMatchSnapshot();
  expect(screen.getByText("DEV")).toBeInTheDocument();
  expect(screen.getByText("Development Environment")).toBeInTheDocument();
});

test("it renders a yellow variant of the Phase Banner", async () => {
  const { container } = render(
    <PhaseBanner tag="STAGING" variant="yellow">
      <span>Staging Environment</span>
    </PhaseBanner>,
  );

  expect(container).toMatchSnapshot();
  expect(screen.getByText("STAGING")).toBeInTheDocument();
  expect(screen.getByText("Staging Environment")).toBeInTheDocument();
});

test("it renders a green variant of the Phase Banner", async () => {
  const { container } = render(
    <PhaseBanner tag="BETA" variant="green">
      <span>Production / Beta Environment</span>
    </PhaseBanner>,
  );

  expect(container).toMatchSnapshot();
  expect(screen.getByText("BETA")).toBeInTheDocument();
  expect(screen.getByText("Production / Beta Environment")).toBeInTheDocument();
});

test("it renders a red variant of the Phase Banner", async () => {
  const { container } = render(
    <PhaseBanner tag="TEST" variant="green">
      <span>Test Environment</span>
    </PhaseBanner>,
  );

  expect(container).toMatchSnapshot();
  expect(screen.getByText("TEST")).toBeInTheDocument();
  expect(screen.getByText("Test Environment")).toBeInTheDocument();
});
