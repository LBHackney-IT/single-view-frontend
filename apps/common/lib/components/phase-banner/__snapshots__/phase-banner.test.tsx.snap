// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a default blue variant of the Phase Banner 1`] = `
<div>
  <div
    class="container-max-width lbh-phase-banner"
  >
    <p
      class="govuk-phase-banner__content"
    >
      <strong
        class="lbh-tag--grey govuk-phase-banner__content__tag govuk-tag lbh-tag"
      >
        DEV
      </strong>
      <span
        class="govuk-phase-banner__text"
      >
        <span>
          Development Environment
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[`it renders a green variant of the Phase Banner 1`] = `
<div>
  <div
    class="container-max-width lbh-phase-banner"
  >
    <p
      class="govuk-phase-banner__content"
    >
      <strong
        class="lbh-tag--green govuk-phase-banner__content__tag govuk-tag lbh-tag"
      >
        BETA
      </strong>
      <span
        class="govuk-phase-banner__text"
      >
        <span>
          Production / Beta Environment
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[`it renders a red variant of the Phase Banner 1`] = `
<div>
  <div
    class="container-max-width lbh-phase-banner"
  >
    <p
      class="govuk-phase-banner__content"
    >
      <strong
        class="lbh-tag--green govuk-phase-banner__content__tag govuk-tag lbh-tag"
      >
        TEST
      </strong>
      <span
        class="govuk-phase-banner__text"
      >
        <span>
          Test Environment
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[`it renders a yellow variant of the Phase Banner 1`] = `
<div>
  <div
    class="container-max-width lbh-phase-banner"
  >
    <p
      class="govuk-phase-banner__content"
    >
      <strong
        class="lbh-tag--yellow govuk-phase-banner__content__tag govuk-tag lbh-tag"
      >
        STAGING
      </strong>
      <span
        class="govuk-phase-banner__text"
      >
        <span>
          Staging Environment
        </span>
      </span>
    </p>
  </div>
</div>
`;
